src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala

```diff
@@ -17,20 +17,28 @@ package za.co.absa.hyperdrive.trigger.api.rest.controllers
 
 import java.util.concurrent.CompletableFuture
 
-import za.co.absa.hyperdrive.trigger.api.rest.services.WorkflowService
-import za.co.absa.hyperdrive.trigger.models.{Project, ProjectInfo, Workflow, WorkflowJoined, WorkflowState}
 import javax.inject.Inject
 import org.springframework.web.bind.annotation._
+import za.co.absa.hyperdrive.trigger.api.rest.services.WorkflowService
+import za.co.absa.hyperdrive.trigger.models.errors.{ApiError, ApiException}
+import za.co.absa.hyperdrive.trigger.models._
 
-import scala.concurrent.ExecutionContext.Implicits.global
 import scala.compat.java8.FutureConverters._
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
 
 @RestController
 class WorkflowController @Inject()(workflowService: WorkflowService) {
 
+  implicit def eitherToCompletableFutureOrException[T](response: Future[Either[Seq[ApiError], T]]): CompletableFuture[T] =
```

Comment:
I would move this to Controller interface/class and implements it here and we can extract .toJava.toCompletableFuture as well

Reply:
As discussed, won't do now